llvm/include/llvm/Transforms/IPO/DeadArgumentElimination.h

```diff
@@ -123,9 +124,9 @@ class DeadArgumentEliminationPass
   /// (used only by bugpoint).
   bool ShouldHackArguments = false;
 
-  /// This allows to eliminate dead arguments in SPIR kernel functions with
-  /// external linkage in SYCL environment
-  bool CheckSpirKernels = false;
+  /// This allows to eliminate dead arguments in SYCL kernel functions with
+  /// external linkage
```

Comment:
Sorry for being annoying with a stupid things like this, but technically speaking the function we process here is not "SYCL kernel function".
[SYCL 2020](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html) defines "SYCL kernel function" as 

> A type which is callable with operator() that takes a id, item, nd-item or work-group which can be passed to kernel enqueue member functions of the command group handler. A SYCL kernel function defines an entry point to a kernel. The function object can be a named device copyable type or lambda function.

Here we process a function wrapping SYCL kernel (SYCL kernel wrapper function?), which SYCL compiler generates to address OpenCL (and probably other back-end's) restrictions for "kernel functions".

I wish we have a good name for this wrapper function. Originally we used OpenCL kernel function, but renamed to avoid an impression that it's written in OpenCL language. Obviously SPIR kernel function name has it's own issues.

Any ideas for a better name?

Reply:
No problem - good to be precise. I can't think of anything more concise than "SYCL kernel wrapper function", so shall I just change it to that?

Reply:
Unless someone else has a better idea, I'm find with that change.